h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -1188,6 +1282,33 @@ private Chunk chk_xnew(Chunk[] chks, int c) {
       }
     }
 
+
+    /* same as ArrayUtils.multVecArr() but faster I hope. */
+    private double[] multVecArrFast(double[] xnew, Archetypes yt, int j) {
+      double[] xy = new double[yt._numLevels[j]];
+      if (yt._transposed) {
+        for (int level = 0; level < yt._numLevels[j]; level++) {
+          int cidx = yt.getCatCidx(j, level);
+          for (int k = 0; k < _ncolX; k++) {
+            double archValue = 0.0;
+            archValue = yt._archetypes[cidx][k];
```

Comment:
`yt._archetypes[cidx]` can be computed in the outer loop
Essentially this entire inner loops simply computes dot-product between `xnew` and `yt._archetypes[cidx]` -- there's probably even a function in MathUtils that you can reuse for this purpose